docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -153,7 +157,711 @@ Future releases
+7. MeshTrust should be removed by the user
+
+![Rotation flow](assets/073/rotation-flow.png)
+
+#### Secret naming
+
+We want to use a [KRI](070-resource-identifier.md) for resource naming. Since `MeshIdentity` creates a separate resource (Secret), it should be named based on the KRI to simplify correlation between the `Secret` and the `MeshIdentity` configuration.
+
+Proposed name:
+
+```yaml
+kri_mi_mesh-1_us-east-2_kuma-system_identity-1
+```
+
+Also, we need to name the Spire agent cluster, in this case is it internal or maybe based on MeshIdentity:
+Proposed name:
+```yaml
+system_spire-agent
+```
+
+or 
+```yaml
+kri_mi_mesh-1_us-east-2_kuma-system_spire-identity
+```
+
+> [!WARNING]
+> The naming is different when using SPIRE, since SPIRE uses SPIFFEID or `default` as a certificate name
+> https://spiffe.io/docs/latest/deploying/spire_agent/#sds-configuration
+> https://spiffe.io/docs/latest/microservices/envoy/#tls-certificates
+#### MeshTrust
+
+Based on the existing `MeshIdentity`, we should create a `MeshTrust` resource, which will be used to properly generate trust domains for other zones.
+`MeshTrust` should be created in each zone and then synchronized with the global control plane.
+1. Can be created on the Zone and synced to Global
+2. Can be created on the Global and synced to all Zones
+3. Can be created on the Zone and not synced to global by providing label `kuma.io/kds-sync: disabled`
+4. Can be created on the Zone and synced to all zones (Future release) (`SyncedAcrossZonesFlag`)
+
+We can describe it as KDS flag:
+
+```golang
+model.GlobalToZonesFlag | model.ZoneToGlobalFlag
+```
+
+There should also be an option to disable automatic generation in cases where the user does not want cross trust-domain traffic to be allowed by default.
+
+**Model**
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrust
+metadata:
+  name: trust-hash
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  ca: inline | secret | file
+  trustDomain: "prod.zone-1.id.kuma.io"
+```
+
+The resource should also be manually creatable by the user. This enables users to define additional trust domains and allows the local sidecar to accept incoming mTLS traffic from services outside the mesh.
+This functionality is essential for:
+
+* Enabling ExternalClients,
+* Accepting mTLS connections from external services, and
+* Federating multiple separate clusters through shared trust.
+
+**MeshTrust based on MeshIdentity**
+
+When a user creates a `MeshIdentity`, we will automatically create a corresponding `MeshTrust` resource using the CA and trust domain from that identity. We can implement a dedicated generator that creates the MeshTrust based on the `MeshIdentity`.
+To avoid issues where a user removes a `MeshIdentity` but an old trust remains in use, we will not remove a `MeshTrust` and we let user to clean it up. In case it's problematic for the user we can implement it later.
```

Comment:
What happens with updates to MeshIdentity or deletion/recreation? Is the MeshTrust replaced? Updated with both?